test/prerequisite-tasks.js

```diff
@@ -0,0 +1,247 @@
+			exitCode: 0,
+			stdout: ''
+		}
+	]);
+	await run(testedModule('2.0.0', {name: 'test', version: '1.0.0', private: true}, {yarn: false}));
+	t.true(tasks.some(task => task.title === 'Ping npm registry' && task.isSkipped()));
+});
+
+test.serial('external registry: should skip task pinging npm registry', async t => {
+	execaStub.createStub([
+		{
+			command: 'git rev-parse --quiet --verify refs/tags/v2.0.0',
+			exitCode: 0,
+			stdout: ''
+		}
+	]);
+	await run(testedModule('2.0.0', {name: 'test', version: '1.0.0', publishConfig: {registry: 'http://my.io'}},
+		{yarn: false}));
+	t.true(tasks.some(task => task.title === 'Ping npm registry' && task.isSkipped()));
+});
+
+test.serial('should fail when npm version does not match range in `package.json`', async t => {
+	execaStub.createStub([
+		{
+			command: 'npm --version',
+			exitCode: 0,
+			stdout: '6.0.0'
+		},
+		{
+			command: 'git rev-parse --quiet --verify refs/tags/v2.0.0',
+			exitCode: 0,
+			stdout: ''
+		}
+	]);
+	const depRange = require('../package.json').engines.npm;
+	await t.throwsAsync(run(testedModule('2.0.0', {name: 'test', version: '1.0.0'}, {yarn: false})),
+		{message: `Please upgrade to npm${depRange}`});
+	t.true(tasks.some(task => task.title === 'Check npm version' && task.hasFailed()));
+});
+
+test.serial('should fail when yarn version does not match range in `package.json`', async t => {
+	execaStub.createStub([
+		{
+			command: 'yarn --version',
+			exitCode: 0,
+			stdout: '1.0.0'
+		},
+		{
+			command: 'git rev-parse --quiet --verify refs/tags/v2.0.0',
+			exitCode: 0,
+			stdout: ''
+		}
+	]);
+	const depRange = require('../package.json').engines.yarn;
+	await t.throwsAsync(run(testedModule('2.0.0', {name: 'test', version: '1.0.0'}, {yarn: true})),
+		{message: `Please upgrade to yarn${depRange}`});
+	t.true(tasks.some(task => task.title === 'Check yarn version' && task.hasFailed()));
+});
+
+test.serial('should fail when user is not authenticated at npm registry', async t => {
+	execaStub.createStub([
+		{
+			command: 'npm whoami',
+			exitCode: 0,
+			stdout: 'sindresorhus'
+		},
+		{
+			command: 'npm access ls-collaborators test',
+			exitCode: 0,
+			stdout: '{"sindresorhus": "read"}'
+		}
+	]);
+	process.env.NODE_ENV = 'P';
+	await t.throwsAsync(run(testedModule('2.0.0', {name: 'test', version: '1.0.0'}, {yarn: false})),
+		{message: 'You do not have write permissions required to publish this package.'});
+	process.env.NODE_ENV = 'test';
+	t.true(tasks.some(task => task.title === 'Verify user is authenticated' && task.hasFailed()));
+});
+
+test.serial('should fail when user is not authenticated at external registry', async t => {
+	execaStub.createStub([
+		{
+			command: 'npm whoami --registry http://my.io',
+			exitCode: 0,
+			stdout: 'sindresorhus'
+		},
+		{
+			command: 'npm access ls-collaborators test --registry http://my.io',
+			exitCode: 0,
+			stdout: '{"sindresorhus": "read"}'
+		}
+	]);
+	process.env.NODE_ENV = 'P';
```

Comment:
Setting the NODE_ENV-environment-variable was necessary. Otherwise this task is always skipped
(see https://github.com/sindresorhus/np/blob/master/source/.prerequisite-tasks.js line 33).
When i put this environment-variable directly in ava-configuration, some other were broken.